app/models/category.rb

```diff
@@ -859,24 +883,9 @@ def self.find_by_email(email)
     self.where("string_to_array(email_in, '|') @> ARRAY[?]", Email.downcase(email)).first
   end
 
-  @@has_children = DistributedCache.new("has_children")
-
-  def self.has_children?(category_id)
-    @@has_children.defer_get_set(category_id.to_s) do
-      Category.where(parent_category_id: category_id).exists?
-    end
-  end
-
   def has_children?
-    !!id && Category.has_children?(id)
-  end
-
-  def self.clear_parent_ids
-    @@has_children.clear
-  end
-
-  def clear_parent_ids
-    Category.clear_parent_ids
+    @has_children ||= (id && Category.where(parent_category_id: id).exists?) ? :true : :false
```

Comment:
Is there a reason we no longer cache this in a `DistributedCache`? I'm also wondering if this should be a column instead because the fact that a category has child categories or not does not change often.

Reply:
> if the current user can see or not any of the children categories. 

Looking at the query, it doesn't seem like this condition is present though. If the scope is important, then the instance variable cache will have to be keyed on some user specific attribute as well.

Reply:
Ooo icic. This method is no longer used in the serializer.